cmds/modules/noded/main.go

```diff
@@ -245,6 +247,50 @@ func action(cli *cli.Context) error {
 		}
 	}()
 
+	// monitor node env updates in substrate url
+	go func() {
+		for {
+			<-time.After(10 * time.Minute)
+
+			log.Debug().Msg("checking if node is maintaining a healty substrate connection")
+			cl, _, err := sub.Raw()
+			if err == nil {
+				// skip update if the connection is working properly
+				log.Debug().Msg("the open connection is healty, no update needed")
+				cl.Client.Close()
+				continue
+			}
+			log.Debug().Err(err).Msg("used substrate manager is not healty, trying to update it")
+
+			// update only if the substrate connection is not healty
+			newEnv, err := environment.Get()
+			if err != nil {
+				log.Debug().Err(err).Msg("failed to get updated config")
+				continue
+			}
+
+			newSubURLs := newEnv.SubstrateURL
+			slices.Sort(subURLs)
+			slices.Sort(newSubURLs)
+
+			if slices.Equal(subURLs, newSubURLs) {
+				log.Debug().Msg("zos-config doesn't have updated config to update the node with")
+				continue
+			}
+
+			newSub, err := environment.GetSubstrate()
```

Comment:
correct me, i see GetSubstrate do the exact checks above? fetch urls, assure urls changed, check the connection, and neturn new manager